### PR TITLE
fix 'pod' in kubelet prober metrics

### LINKED ISSUE #115766: Metric prober_probe_total reports invalid pod field
### What happened?

The metric `prober_probe_total` used to report acutal pod in the label `pod`, but now I see the podname prefix only.

### What did you expect to happen?

The full pod name in the label, so I can trace the metric to the failing pod.

### How can we reproduce it (as minimally and precisely as possible)?

Check raw api
```
$ kubectl get --raw /api/v1/nodes/$(kubectl get node -o jsonpath='{.items[1].metadata.name}')/proxy/metrics/probes  |grep "probe_total.*haprox"
prober_probe_total{container="haproxy-ingress",namespace="haproxy-controller",pod="haproxy-kubernetes-ingress",pod_uid="a8230fd3-a53a-4b22-a967-0ed7ffa25740",probe_type="Liveness",result="successful"} 59948
```
```
$ kubectl get --raw /api/v1/nodes/$(kubectl get node -o jsonpath='{.items[0].metadata.name}')/proxy/metrics/probes  |grep "probe_total.*haprox"
prober_probe_total{container="haproxy-ingress",namespace="haproxy-controller",pod="haproxy-kubernetes-ingress",pod_uid="4538d5ba-33a8-4312-95cd-c9f12a955ddb",probe_type="Liveness",result="successful"} 59969
```
and verify the actual pod name with uids.
```
$ kubectl -n haproxy-controller get pods -o json | jq -r '.items[].metadata|([.name,.uid])|@tsv'
haproxy-kubernetes-ingress-default-backend-69669bc5c5-fmb8w	f769c7a2-beb5-4375-974f-a6cc4489697a
haproxy-kubernetes-ingress-dslfv	e0139316-6fc0-4980-a3b6-f553ad39efb0
haproxy-kubernetes-ingress-tz9kt	9f260c46-99d4-4e88-a521-494acf3125df
haproxy-kubernetes-ingress-vx24d	1a7ae6d4-911b-4e55-8422-2e631b8ef355
haproxy-kubernetes-ingress-zkf5g	3573c577-9410-43a4-9b23-599f9bcbdc2b
```

### Anything else we need to know?

https://github.com/kubernetes/kubernetes/blob/71ec5ed81df288a8bd4c1c6a89df404d30ed0cd3/pkg/kubelet/prober/prober_manager.go#L46

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:16:20Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.5", GitCommit:"804d6167111f6858541cef440ccc53887fbbc96a", GitTreeState:"clean", BuildDate:"2022-12-11T00:17:11Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.25) exceeds the supported minor version skew of +/-1
```

</details>

### Cloud provider

Azure AKS

### OS version

<details>

```console
$ kubectl get node -o json| jq '.items[0].status.nodeInfo|del(.machineID,.systemUUID,.bootID)'
{
  "architecture": "amd64",
  "containerRuntimeVersion": "containerd://1.6.15+azure-1",
  "kernelVersion": "5.15.0-1031-azure",
  "kubeProxyVersion": "v1.25.5",
  "kubeletVersion": "v1.25.5",
  "operatingSystem": "linux",
  "osImage": "Ubuntu 22.04.1 LTS"
}
```

</details>

### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by a7i**
#### What type of PR is this?
/kind bug
/kind regression

#### Which issue(s) this PR fixes:

Fixes #115766

#### Special notes for your reviewer:

Kubernetes 1.25 Metric:
```
prober_probe_total{container="app-nginx",namespace="app",
pod="app-nginx",pod_uid="2de634c5-963f-4ffb-9b15-700f3f9b8bf3",
probe_type="Readiness",result="successful"
} 794
```

Kubernetes 1.24 Metric:
```
prober_probe_total{container="app-nginx",namespace="app",
pod="app-nginx-799b74869b-mmhr7",pod_uid="4550420b-6a0c-4d59-a486-f5b3a69ebdf8",
probe_type="Readiness",result="successful"
} 81023
```

This breaking change causes the metrics to not get processed by [datadog core](https://github.com/DataDog/integrations-core/blob/22899496fd0908344f1db0920d55554645b85578/kubelet/datadog_checks/kubelet/probes.py#L94) which relies on actual pod name.

#### Does this PR introduce a user-facing change?
Yes, but it reverts to initial functionality.

```release-note
Fixes a 1.25 regression to revert kubelet prober metrics `pod` tag to include actual pod name
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @jackfrancis  @SergeyKanzhelev from discussion [here](https://github.com/kubernetes/kubernetes/issues/115766#issuecomment-1446747588)

### CODE REVIEW

**Comment by mindw** [comment]
Please consider back porting this fix supported releases - 1.25+. Thanks!

**Comment by dosmanak** [comment]
Hi, thank you for making this simple revert.  I wonder why the issue has not been yet targeted by the core team as it seems clear.
I hope in soon merge in the main and possible backport in current release.

**Comment by derekwaynecarr** [comment]
I think we should take the revert and return to prior behavior given the regression.

Would like @SergeyKanzhelev to chime in as well since he reviewed prior PR.

/lgtm

**Comment by derekwaynecarr** [comment]
pending review from a few others.

/hold

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

The concern was the cardinality explosion as pod name has replica's random part. However we are not making things worse exponentially - we already had a metric for success. 

I think the right long term fix will be having this logic as a default, but also allow to override the app name in those metrics based on some attributes

**Comment by dims** [comment]
/approve
/lgtm

per discussion on slack and here, we need to get this in.

**Pull request merged**
